editor/utils/auth.ts

```diff
@@ -78,9 +79,15 @@ export const useGetToken = (): (() => Promise<string | undefined>) => {
       const accessToken = await getAccessTokenSilently();
       return accessToken;
     } catch (err) {
-      console.error("Error getting access token:", err);
+      console.warn(
+        "Redirecting to login page due to error getting access token:",
+        err
+      );
+      loginWithRedirect({
+        appState: { returnTo: window.location.pathname },
+      });
```

Comment:
Here's the missing error handler. If we can't get a new token (most likely because the user's refresh token has expired) we route the user to the login page and save their current window location in state. 

The `appState` payload gets sent to the Auth0 sign in page, and Auth0 sends it back to the client after sign-in by encoding it in the app's URL. The new `onRedirectCallback` in `Auth.tsx` handles this payload.

Reply:
very nice! cool to see that there is an clean way to pass the previous route. 🙌 ✨